### PR TITLE
Mark style as invalid when new style is loaded and apply it to location component

### PR DESCRIPTION
**Pull request by kiryldz**
### Summary of changes

Treat `Style` object as invalid when new style is loaded. Accessing invalid object will log errors.
As a result, location component plugin was revisited to use correct (latest loaded) style.

Another plugin that is dependent on style change - annotation plugin - was not touched as it always uses latest style by calling `MapDelegateProvider.getStyle { }` and not storing direct references to any `Style` objects.

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [x] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [x] Document any changes to public APIs.
 - [ ] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail.
 - [x] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: sdk/src/main/java/com/mapbox/maps/StyleObserver.kt [hunk 1]
```diff
@@ -26,7 +26,8 @@ internal class StyleObserver(
   private var loadStyleTransitionOptions: TransitionOptions? = null
 
   private val getStyleListeners = CopyOnWriteArraySet<Style.OnStyleLoaded>()
-  private val loadedStyleList = CopyOnWriteArraySet<Style>()
+  private var loadedStyle: Style? = null
```
Comment (kiryldz) [inline sdk/src/main/java/com/mapbox/maps/StyleObserver.kt:29]: @mapbox/maps-android that's actually core change introduced by this PR.

File: plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationIndicatorLayerRenderer.kt [hunk 2]
```diff
@@ -92,6 +93,11 @@ internal class LocationIndicatorLayerRenderer(
     style?.removeStyleImage(SHADOW_ICON)
   }
 
+  override fun updateStyle(style: StyleInterface) {
+    this.style = style
```
Comment (alexander-kulikovskii) [inline plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationIndicatorLayerRenderer.kt:97]: can we compare with previous version?

Comment (kiryldz) [reply plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationIndicatorLayerRenderer.kt:97]: we have basically a lot of occurrences in this PR but all of them are coming from a single place and it's basically `StyleObserver` where we create the new object everytime style is loaded. to sum up I see no sense in adding those checks, imo it simply makes code a bit more polluted

Comment (alexander-kulikovskii) [reply plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationIndicatorLayerRenderer.kt:97]: oh, sorry
didn't notice that it happened inside internal class. thought that user can set this style.

Comment (kiryldz) [reply plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationIndicatorLayerRenderer.kt:97]: no, all the work done in this PR is not exposed to an end user

**Review by alexander-kulikovskii — APPROVED** [review]

File: plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/ModelSourceWrapper.kt [hunk 3]
```diff
@@ -28,8 +29,12 @@ internal class ModelSourceWrapper(
     sourceProperties[MODELS] = Value(models)
   }
 
+  fun updateStyle(style: StyleInterface) {
+    this.style = style
+  }
+
   fun bindTo(delegate: StyleManagerInterface) {
```
Comment (yunikkk) [inline plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/ModelSourceWrapper.kt:36]: rename param to `style` as well?

File: sdk/src/main/java/com/mapbox/maps/Style.kt [hunk 4]
```diff
@@ -27,14 +27,14 @@ class Style internal constructor(
 
   private var isStyleValid = true
 
-  internal fun onMapViewDestroyed() {
+  internal fun markStyleInvalid() {
```
Comment (yunikkk) [inline sdk/src/main/java/com/mapbox/maps/Style.kt:30]:
```suggestion
  internal fun markInvalid() {
```

**Review by yunikkk — APPROVED** [review]

**Review by pengdev — COMMENTED** [review]
LGTM, let's keep testing these changes during the beta and rc phases, hopefully there's no significant impact or minor bugs to end user.

**Comment by kiryldz** [comment]
> LGTM, let's keep testing these changes during the beta and rc phases, hopefully there's no significant impact or minor bugs to end user.

Particularly this change does not bring in any behavioural changes, even using old style reference will work perfectly fine but logs will be written.

**Pull request merged**